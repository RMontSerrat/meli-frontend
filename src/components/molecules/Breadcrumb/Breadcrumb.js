import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import AngleRight from '../../../assets/icons/angle-right.svg';
import styles from './Breadcrumb.module.scss';

const Breadcrumb = ({ items }) => (
  <ol className={styles.Breadcrumb} data-testid="Breadcrumb">
    {items.map((item, i) => (
      <Fragment key={item}>
        <li>
          {item}
        </li>
        {i < items.length - 1 && <AngleRight />}
      </Fragment>
    ))}
  </ol>
);

Breadcrumb.propTypes = {
  items: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default Breadcrumb;
