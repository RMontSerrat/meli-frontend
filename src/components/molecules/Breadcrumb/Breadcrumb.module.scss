$color: color(dark, base);

.Breadcrumb {
  align-items: center;
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-start;
  margin: 0;
  padding: 10px 0;

  li {
    color: color(dark, base);
    display: inline;
    font-family: $font-family;
    font-size: size(small);
    font-weight: 300;
    line-height: normal;
  }

  svg {
    padding: 0 5px;
    width: 7px;

    path {
      fill: $color;
    }
  }
}
