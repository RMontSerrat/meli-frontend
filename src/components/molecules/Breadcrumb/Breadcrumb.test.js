import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Breadcrumb from './Breadcrumb';

const defaultItems = ['Vestuário Masculino', 'Calçados', 'Tênis'];

describe('<Breadcrumb />', () => {
  test('it should mount', () => {
    render(<Breadcrumb items={defaultItems} />);

    const breadcrumb = screen.getByTestId('Breadcrumb');

    expect(breadcrumb).toBeInTheDocument();
  });

  test('snapshot', () => {
    const { asFragment } = render(
      <Breadcrumb items={defaultItems} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
