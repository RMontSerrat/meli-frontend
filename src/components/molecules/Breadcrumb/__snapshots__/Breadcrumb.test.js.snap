// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb /> snapshot 1`] = `
<DocumentFragment>
  <ol
    class="Breadcrumb"
    data-testid="Breadcrumb"
  >
    <li>
      Vestuário Masculino
    </li>
    <svg
      class="svg-inline--fa fa-angle-right fa-w-6 fa-3x"
      data-icon="angle-right"
      data-prefix="fal"
      viewBox="0 0 192 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M166.9 264.5l-117.8 116c-4.7 4.7-12.3 4.7-17 0l-7.1-7.1c-4.7-4.7-4.7-12.3 0-17L127.3 256 25.1 155.6c-4.7-4.7-4.7-12.3 0-17l7.1-7.1c4.7-4.7 12.3-4.7 17 0l117.8 116c4.6 4.7 4.6 12.3-.1 17z"
        fill="currentColor"
      />
    </svg>
    <li>
      Calçados
    </li>
    <svg
      class="svg-inline--fa fa-angle-right fa-w-6 fa-3x"
      data-icon="angle-right"
      data-prefix="fal"
      viewBox="0 0 192 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M166.9 264.5l-117.8 116c-4.7 4.7-12.3 4.7-17 0l-7.1-7.1c-4.7-4.7-4.7-12.3 0-17L127.3 256 25.1 155.6c-4.7-4.7-4.7-12.3 0-17l7.1-7.1c4.7-4.7 12.3-4.7 17 0l117.8 116c4.6 4.7 4.6 12.3-.1 17z"
        fill="currentColor"
      />
    </svg>
    <li>
      Tênis
    </li>
  </ol>
</DocumentFragment>
`;
